### PR TITLE
Add new `Thread.archive`

### PR DESCRIPTION
**Pull request by Sengolda**
## Summary
Adds a new Thread.archive method that archives a thread.

## Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting, examples, ...)

### CODE REVIEW

File: discord/threads.py [hunk 1]
```diff
@@ -589,6 +589,25 @@ async def edit(
         data = await self._state.http.edit_channel(self.id, **payload)
         # The data payload will always be a Thread payload
         return Thread(data=data, state=self._state, guild=self.guild)  # type: ignore
+    
+
+    async def archive(self, locked: bool = MISSING) -> Thread:
+        """|coro|
+
+        Used to archive a thread, this is an alias to using :meth:`Thread.edit` to archive a thread.
```
Comment (RPSMain) [inline discord/threads.py:597]:
```suggestion
        Used to archive a thread, aliases :meth:`Thread.edit` to archive a thread.
```

File: discord/threads.py [hunk 2]
```diff
@@ -589,6 +589,25 @@ async def edit(
...
+        Parameters
+        ------------
+        locked: :class:`bool`
+            When set to True it will lock the thread. This will default to False.
+        
+
+        Returns
+        --------
+        :class:`Thread`
+            The thread that you have just archived.
```
Comment (RPSMain) [inline discord/threads.py:608]:
```suggestion
            The thread that you have archived.
```

**Review by RPSMain — CHANGES_REQUESTED** [review]

**Review by Dorukyum — CHANGES_REQUESTED** [review]
There are some wording inconsistencies. See the docstring for `Thread.edit`.

File: discord/threads.py [hunk 3]
```diff
@@ -589,6 +589,25 @@ async def edit(
         data = await self._state.http.edit_channel(self.id, **payload)
         # The data payload will always be a Thread payload
         return Thread(data=data, state=self._state, guild=self.guild)  # type: ignore
+    
+
+    async def archive(self, locked: bool = MISSING) -> Thread:
+        """|coro|
+
+        Used to archive a thread, aliases :meth:`Thread.edit` to archive a thread.
```
Comment (nerdguyahmad) [inline discord/threads.py:597]:
```suggestion
        Archives the thread. This is a shorthand of :meth:`.edit`.
```

File: discord/threads.py [hunk 4]
```diff
@@ -589,6 +589,25 @@ async def edit(
...
+
+        Parameters
+        ------------
+        locked: :class:`bool`
+            When set to True it will lock the thread. This will default to False.
```
Comment (nerdguyahmad) [inline discord/threads.py:602]:
```suggestion
            Whether to lock the thread on archive, Defaults to False.
```

File: discord/threads.py [hunk 5]
```diff
@@ -589,6 +589,25 @@ async def edit(
...
+        
+
+        Returns
+        --------
+        :class:`Thread`
+            The thread that you have archived.
```
Comment (nerdguyahmad) [inline discord/threads.py:608]:
```suggestion
            The updated thread.
```

**Review by nerdguyahmad — CHANGES_REQUESTED** [review]

**Review by VincentRPS — DISMISSED** [review]

File: discord/threads.py [hunk 6]
```diff
@@ -589,6 +589,25 @@ async def edit(
         data = await self._state.http.edit_channel(self.id, **payload)
         # The data payload will always be a Thread payload
         return Thread(data=data, state=self._state, guild=self.guild)  # type: ignore
+    
+
```
Comment (Lulalaby) [inline discord/threads.py:593]:
```suggestion
```

File: discord/threads.py [hunk 7]
```diff
@@ -589,6 +589,25 @@ async def edit(
...
+    async def archive(self, locked: bool = MISSING) -> Thread:
```
Comment (Lulalaby) [inline discord/threads.py:594]: Add a unarchive function maybe?

**Review by Lulalaby — CHANGES_REQUESTED** [review]

File: discord/threads.py [hunk 8]
```diff
@@ -589,6 +589,42 @@ async def edit(
...
+        
+
+        Returns
+        --------
+        :class:`Thread`
+            The updated thread.
+        """
+        return await self.edit(archived=True, locked=locked)
+    
+    async def unarchive(self, locked: bool = MISSING) -> Thread:
```
Comment (Lulalaby) [inline discord/threads.py:611]: remove locked :)

**Review by Lulalaby — DISMISSED** [review]

File: discord/threads.py [hunk 9]
```diff
@@ -589,6 +589,36 @@ async def edit(
...
+        return await self.edit(archived=True, locked=locked)
+    
+    async def unarchive(self) -> Thread:
+        """|coro|
+
+        Unarchives the thread. This is a shorthand of :meth:`.edit`.
+
+        Returns
+        --------
+        :class:`Thread`
```
Comment (Dorukyum) [inline discord/threads.py:618]:
```suggestion
        :class:`.Thread`
```

File: discord/threads.py [hunk 10]
```diff
@@ -589,6 +589,36 @@ async def edit(
...
+
+        Parameters
+        ------------
+        locked: :class:`bool`
+            Whether to lock the thread on archive, Defaults to ``False``.
+        
+
+        Returns
+        --------
+        :class:`Thread`
```
Comment (Dorukyum) [inline discord/threads.py:606]:
```suggestion
        :class:`.Thread`
```

**Review by Dorukyum — CHANGES_REQUESTED** [review]

**Review by Lulalaby — DISMISSED** [review]

**Review by Dorukyum — APPROVED** [review]

**Review by Lulalaby — APPROVED** [review]

**Pull request merged**